ddl/ddl_api.go

```diff
@@ -3366,10 +3366,66 @@ func (d *ddl) UnlockTables(ctx sessionctx.Context, unlockTables []model.TableLoc
 	return errors.Trace(err)
 }
 
+func (d *ddl) CleanupTableLock(ctx sessionctx.Context, tables []*ast.TableName) error {
+	uniqueTableID := make(map[int64]struct{})
+	cleanupTables := make([]model.TableLockTpInfo, 0, len(tables))
+	// Check whether the table was already locked by other.
+	for _, tb := range tables {
+		// TODO: replace const string "performance_schema" with xxx.LowerName.
+		// Currently use perfschema.LowerName will have import cycle problem.
+		if tb.Schema.L == infoschema.LowerName || tb.Schema.L == "performance_schema" || tb.Schema.L == mysql.SystemDB {
```

Comment:
I think you better to use a function for this check, I found you repeat the code in https://github.com/pingcap/tidb/blob/a935d93990ad7059d3f2a07eeec1eda71f257353/ddl/ddl_api.go#L3401